runtime/src/main/java/org/corfudb/runtime/collections/TransactionPoller.java

```diff
@@ -69,18 +83,12 @@ private void pollTxStream() {
  
         long lastReadAddress = streamContexts.get(0).getLastReadAddress();
 
-        StreamOptions options = StreamOptions.builder()
```

Comment:
I don't think having the try/catch in run is a good idea. Maybe only catching a trimmed exception is better.

Based on the comment, I think the problem is if you pass a runnable to a scheduler executor and it ends up throwing an exception, then the second run won't schedule that runnable to run. I think its cleaner for the executor owner to actually wrap this runnable with a try/catch and log the exception.

Reply:
Tracking this comment, for addressing in an upcoming PR. thanks